### PR TITLE
hibernate-validator: better support for custom ConstraintValidatorFactory

### PR DESCRIPTION
**Pull request by jknack**
- simplify custom ConstraintValidatorFactory based on DI as well as manually created factories
- integrates Validator factory into hibernate

### CODE REVIEW

**Comment by jknack** [comment]
@kliushnichenko please check

**Review by kliushnichenko — COMMENTED** [review]
Cool! Added my 2 cents to it

File: modules/jooby-hibernate-validator/src/main/java/io/jooby/hibernate/validator/HibernateValidatorModule.java [hunk 1]
```diff
@@ -58,6 +59,7 @@ public class HibernateValidatorModule implements Extension {
   private String title = "Validation failed";
   private boolean disableDefaultViolationHandler = false;
   private boolean logException = false;
+  private List<ConstraintValidatorFactory> factories;
```
[the remark this answers is not in the document]
Comment (jknack) [reply modules/jooby-hibernate-validator/src/main/java/io/jooby/hibernate/validator/HibernateValidatorModule.java:62]: Isn't need it if we rely on DI. So this is for people who doesn't like DI (reflective or not), so they can just do:

```
 getInstance(Class type) {
    if (type == Validator1) {..}
    else if (type== Validator2) {...}
    else return null
  }
```

File: modules/jooby-hibernate-validator/src/main/java/io/jooby/internal/hibernate/validator/RegistryConstraintValidatorFactory.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  private final Registry registry;
+
+  public RegistryConstraintValidatorFactory(Registry registry) {
+    this.registry = registry;
+  }
+
+  @Override
+  public <T extends ConstraintValidator<?, ?>> T getInstance(Class<T> key) {
+    try {
+      return registry.require(key);
```
[the remark this answers is not in the document]
Comment (jknack) [reply modules/jooby-hibernate-validator/src/main/java/io/jooby/internal/hibernate/validator/RegistryConstraintValidatorFactory.java:27]: not sure if possible to implement a registry for Dagger, will add a note.

**Pull request merged**